dm/config/subtask.go

```diff
@@ -47,12 +47,17 @@ const (
 	CmdSyncer CmdName = "syncer"
 )
 
+var (
+	defaultMaxAllowedPacket = 64 * 1024 * 1024 // 64MiB, equal to TiDB's default
+)
+
 // DBConfig is the DB configuration.
 type DBConfig struct {
-	Host     string `toml:"host" json:"host" yaml:"host"`
-	Port     int    `toml:"port" json:"port" yaml:"port"`
-	User     string `toml:"user" json:"user" yaml:"user"`
-	Password string `toml:"password" json:"-" yaml:"password"` // omit it for privacy
+	Host             string `toml:"host" json:"host" yaml:"host"`
+	Port             int    `toml:"port" json:"port" yaml:"port"`
+	User             string `toml:"user" json:"user" yaml:"user"`
+	Password         string `toml:"password" json:"-" yaml:"password"` // omit it for privacy
+	MaxAllowedPacket *int   `toml:"max-allowed-packet" json:"max-allowed-packet" yaml:"max-allowed-packet"`
```

Comment:
why use a `* int` here?

Reply:
differ `0` from _not set_(default)

Reply:
so we allow user to set it to zero, which means no limit to max_allowed_packet ?

Reply:
zero means 
> automatically fetch the max_allowed_packet variable from server on every connection.

from <https://github.com/go-sql-driver/mysql#maxallowedpacket>

Reply:
ok